pymeasure/instruments/rigol/rigol_ds1000z.py

```diff
@@ -0,0 +1,849 @@
+
+    trigger_slope = Instrument.control(
+        ":TRIG:EDGE:SLOP?",
+        ":TRIG:EDGE:SLOP %s",
+        "Control the trigger slope (str): 'POS' or 'NEG'.",
+        validator=strict_discrete_set,
+        values=["POS", "NEG"],
+    )
+
+    trigger_coupling = Instrument.control(
+        ":TRIG:COUP?",
+        ":TRIG:COUP %s",
+        "Control the trigger coupling mode (str).",
+        validator=strict_discrete_set,
+        values=["AC", "DC", "HFRej", "LFRej", "NOIS"],
+    )
+
+    trigger_sweep = Instrument.control(
+        ":TRIG:SWE?",
+        ":TRIG:SWE %s",
+        "Control the trigger sweep mode (str): 'AUTO', 'NORM', or 'SING'.",
+        validator=strict_discrete_set,
+        values=["AUTO", "NORM", "SING"],
+    )
+
+    trigger_level = Instrument.control(
+        ":TRIG:LEV?",
+        ":TRIG:LEV %f",
+        "Control the trigger level in volts (float).",
+        cast=float,
+    )
+
+    waveform_source = Instrument.control(
+        ":WAV:SOUR?",
+        ":WAV:SOUR %s",
+        "Control the waveform source used for data transfers (str).",
+        validator=strict_discrete_set,
+        values=_WAVEFORM_SOURCES,
+    )
+
+    waveform_mode = Instrument.control(
+        ":WAV:MODE?",
+        ":WAV:MODE %s",
+        "Control the waveform acquisition mode for transfer (str).",
+        validator=strict_discrete_set,
+        values=[mode.value for mode in WaveformMode],
+    )
+
+    waveform_format = Instrument.control(
+        ":WAV:FORM?",
+        ":WAV:FORM %s",
+        "Control the waveform data format for transfer (str).",
+        validator=strict_discrete_set,
+        values=[fmt.value for fmt in WaveformFormat],
+    )
+
+    waveform_start = Instrument.control(
+        ":WAV:STAR?",
+        ":WAV:STAR %d",
+        "Control the first data point index for waveform transfer (int).",
+        cast=int,
+    )
+
+    waveform_stop = Instrument.control(
+        ":WAV:STOP?",
+        ":WAV:STOP %d",
+        "Control the last data point index for waveform transfer (int).",
+        cast=int,
+    )
+
+    def run(self):
+        """Start waveform acquisition."""
+        self.write(":RUN")
+
+    def stop(self):
+        """Stop waveform acquisition."""
+        self.write(":STOP")
+
+    def single(self):
+        """Arm the scope for a single acquisition."""
+        self.write(":SING")
+
+    def autoscale(self):
+        """Perform automatic scaling (vertical and timebase)."""
+        self.write(":AUT")
+
+    def clear_status(self):
+        """Clear the instrument status register."""
+        self.write("*CLS")
+
+    def clear(self):
+        """Clear all the waveforms on the screen."""
+        self.write(":CLE")
+
+    def clear_registers(self):
+        """Clear all the event registers and clear the error queue."""
+        self.write(":CLS")
+
+    def reset(self):
+        """Reset the instrument to factory defaults."""
+        self.write("*RST")
+
+    def calibrate_start(self):
```

Comment:
It does not sound that nice. Maybe just `calibrate` (it indicates the action) or `start_calibration`.

The method to stop could be `abort_calibration`.